handlers/handlers.go

```diff
@@ -1,194 +1,127 @@
+        filterString = fmt.Sprintf("%s%s%s", filterString, " AND type = ", patchType) 
 	}
-	// Insert patch datas into database
-	_, err = db.Exec("INSERT INTO patches(timestamp, patch, convo_id, user_id, type) VALUES (?,?,?,?,?) ", patch.Timestamp, patch.Patch, patch.Convo_ID, patch.User_ID, patch.Type);
-	if err != nil {
-		log.Print("Error inserting")
-		log.Print(err)
-		return
+	if (!filter.MaxTime.IsZero()) {
+		filterString = fmt.Sprintf("%s%s%s", filterString, " AND time <= ", filter.MaxTime)
+	}
+	if (!filter.MinTime.IsZero()) {
+		filterString = fmt.Sprintf("%s%s%s", filterString, " AND time >= ", filter.MinTime)
 	}
-	// Send  response
-	response := "Patch created"
-	json.NewEncoder(w).Encode(response)
-
-}
 
-// Read all patches from database
-func GetPatchesHandler(w http.ResponseWriter, r *http.Request) {
-	// Connect to db
-	db, err := dbConnect()
-	defer db.Close()
-	if err != nil {
-		log.Print("Error connecting to database")
-		log.Print(err)
-		return
+	// Get patches with filters
+	if(filterString != ""){
+		filterString = filterString[4:]
 	}
-	// Read table
-	rows, err := db.Query("SELECT * FROM patches")
+	filterString = fmt.Sprintf(" WHERE %s", filterString) 
+	patches, err := env.DB.GetPatches(filterString)
+	
 	if err != nil {
 		log.Print("Error getting rows")
 		log.Print(err)
-		return
-	}
-
-	p := Patch{}
-	patches := make([]Patch, 0)
-	// Create patches object
-	for rows.Next() {
-		err := rows.Scan(&p.Timestamp, &p.Patch, &p.Convo_ID, &p.User_ID, &p.Type)
-		if err != nil {
-			log.Print(err)
-		}
-		patches = append(patches, p)
 	}
 	// Return responses
 	json.NewEncoder(w).Encode(patches)
 }
 
-// Update patch in databsse
-func PutPatchesHandler(w http.ResponseWriter, r *http.Request) {
+// Create new patch in database
+func (env *Env) PostPatchesHandler(w http.ResponseWriter, r *http.Request) {
+
 	// Read http request
 	body, err := ioutil.ReadAll(r.Body)
 	if err != nil {
 		log.Print("Error reading request")
 		log.Print(err)
 	}
-	patch := Patch{}
+	patch := &models.Patch{}
 	err = json.Unmarshal(body, &patch)
 	if err != nil {
 		log.Print("Error reading request")
 		log.Print(err)
 		return
 	}
-	// Connect to db
-	db, err := dbConnect()
-	defer db.Close()
-	if err != nil {
-		log.Print("Error connecting to database")
-		log.Print(err)
-		return
-	}
-	// Update patch in db
-	update, err := db.Exec("UPDATE patches SET patch = ?, convo_id = ?, user_id = ?, type = ? WHERE timestamp = ?;",  patch.Patch, patch.Convo_ID, patch.User_ID, patch.Type, patch.Timestamp)
-	if err != nil {
-		log.Print("Error updating")
-		log.Print(err)
-	}
-	//db.Commit()
-	// Return number of rows affected
-	updated, err := update.RowsAffected()
+
+	err = env.DB.CreatePatch(patch)
+
+	// Send  response
 	if err != nil {
-		log.Print("Error updatin rows")
 		log.Print(err)
-	}
-	if updated > 0 {
-		response := "Rows updated"
-		json.NewEncoder(w).Encode(response)
+		json.NewEncoder(w).Encode(err)
```

Comment:
Use the `http.Error` function to respond with an error. This goes for the other times that the response will be an error.